File: homework2/Python/LTD/dictionary.py
# dictionary 선언
word_count = {
    "apple": 3,
    "banana": 5,
    "cherry": 2
}

# 데이터 출력
print("Word counts:")
for key, value in word_count.items():
    print(f"{key}: {value}")

# 특정 키 검색
key = "banana"
if key in word_count:
    print(f"{key} is found with count: {word_count[key]}")
else:
    print(f"{key} is not found.")

# 데이터 삭제
word_count.pop("cherry", None)

# 삭제 후 출력
print("\nAfter deletion:")
for key, value in word_count.items():
    print(f"{key}: {value}")

File: homework2/Cpp/LTD/Dictionary.cpp
#include <iostream>
#include <map>
#include <string>

int main() {
    // std::map 선언
    std::map<std::string, int> wordCount;

    // 데이터 삽입
    wordCount["apple"] = 3;
    wordCount["banana"] = 5;
    wordCount["cherry"] = 2;

    // 데이터 출력
    std::cout << "Word counts:" << std::endl;
    for (const auto& pair : wordCount) {
        std::cout << pair.first << ": " << pair.second << std::endl;
    }

    // 특정 키 검색
    std::string key = "banana";
    if (wordCount.find(key) != wordCount.end()) {
        std::cout << key << " is found with count: " << wordCount[key] << std::endl;
    } else {
        std::cout << key << " is not found." << std::endl;
    }

    // 데이터 삭제
    wordCount.erase("cherry");

    // 삭제 후 출력
    std::cout << "\nAfter deletion:" << std::endl;
    for (const auto& pair : wordCount) {
        std::cout << pair.first << ": " << pair.second << std::endl;
    }

    return 0;
}